@here/harp-mapview/lib/text/TextElement.ts

```diff
@@ -529,4 +531,14 @@ export class TextElement {
             }
         }
     }
+
+    /**
+     * Disposes of any allocated resources.
+     */
+    dispose() {
+        const poiBuffer = this.poiInfo?.buffer;
+        if (poiBuffer) {
+            poiBuffer.decreaseRefCount();
+        }
+    }
```

Comment:
Decrease ref counting on TextElement disposal to ensure the poi buffer and associated resources are released when ref count reaches 0.